dali/operators/util/reshape.cc

```diff
@@ -132,33 +193,64 @@ void Reshape<Backend>::CalculateOutputShape(const Workspace &ws) {
   const int N = input_shape_.num_samples();
   switch (shape_source_) {
     case ShapeSource::Arg:
-      if (output_shape_.num_samples() != N) {
-        output_shape_ = uniform_list_shape(N, uniform_shape_);
+      if (use_rel_shape_) {
+        output_shape_.resize(N, rel_uniform_shape_.size());
+        for (int i = 0; i < N; i++) {
+          for (int d = 0; d < output_shape_.sample_dim(); d++) {
+            int out_e = round_int(rel_uniform_shape_[d] * input_shape_.tensor_shape_span(i)[d]);
+            output_shape_.tensor_shape_span(i)[d] = out_e;
+          }
+        }
+      } else {
+        if (output_shape_.num_samples() != N) {
+          output_shape_ = uniform_list_shape(N, uniform_shape_);
+        }
       }
       break;
     case ShapeSource::ArgInput:
-      ShapeFromInput(ws.ArgumentInput("shape"));
+      if (use_rel_shape_)
+        ShapeFromInput(ws.ArgumentInput("rel_shape"), true);
+      else
+        ShapeFromInput(ws.ArgumentInput("shape"), false);
       break;
     case ShapeSource::Input:
-      ShapeFromInput(ws.template InputRef<CPUBackend>(1));
+      ShapeFromInput(ws.template InputRef<CPUBackend>(1), false);
       break;
     case ShapeSource::None:
       output_shape_ = input_shape_;
       return;
   }
 
-  DALI_ENFORCE(output_shape_.num_samples() == N,
-    "Reshape: the new shape mush have same number of samples. Got " +
-    to_string(output_shape_.num_samples()) + ", expected " + to_string(N));
+  if (shape_source_ != ShapeSource::None) {
+    for (int i = 0; i < N; i++) {
+      auto actual_volume = volume(input_shape_.tensor_shape_span(i));
+      auto out_sample_shape = output_shape_.tensor_shape_span(i);
+      int wildcard_dim = wildcard_dim_;
+      if (shape_source_ != ShapeSource::Arg) {
```

Comment:
```suggestion
      if (shape_source_ ==ShapeSource::Input) {
```
ShapeSource::ArgInput has been validated already,  ShapeSource::None should by valid by definition.

Reply:
It can also be ArgInput.